samcli/lib/docker/log_streamer.py

```diff
@@ -0,0 +1,96 @@
+"""
+Log streaming utilities when streaming logs from Docker
+"""
+import os
+from typing import Dict
+
+import docker
+from docker.errors import APIError
+
+from samcli.lib.package.stream_cursor_utils import (
+    CursorUpFormatter,
+    CursorDownFormatter,
+    CursorLeftFormatter,
+    ClearLineFormatter,
+)
+from samcli.lib.utils.stream_writer import StreamWriter
+
+
+class LogStreamer:
+    def __init__(self, stream: StreamWriter, error_class: APIError.__class__, error_msg_prefix=""):
```

Comment:
Why do we pass in an exception type here instead of raising a static one, catching it, and throwing whatever is needed in the caller?
Typing might also be wrong, `APIError` is from `docker.errors` but `DockerPushFailedError` is from `UserException`.
For passing class types, `typing.Type[APIError]` should be used.